vcpkg/users/binarycaching-troubleshooting.md

```diff
@@ -0,0 +1,164 @@
+---
+title: Binary caching troubleshooting guide
+description: Troubleshooting your binary caching setup
+author: dan-shaw
+ms.author: danshaw2
+ms.date: 9/20/2023
+ms.prod: vcpkg
+---
+
+# Binary caching troubleshooting guide
+
+This guide is intended for vcpkg binary caching users that are running into unexpected behavior from this experience. Please review the [binary caching](./binarycaching.md) reference documentation for a basic understanding of what this feature does and how it works.
+
+## Set up vcpkg to print debugging information
+
+Debug mode outputs information such as the compiler verison and ABI hashes, which will help us to diagnose any binary caching issues.
+
+* [Classic mode](./classic-mode.md): add `--debug` to your command invocation.
+* CMake toolchain: set `-DVCPKG_INSTALL_OPTIONS="--debug"` in your `cmake` command call or in your `CMakePresets.json` file.
+* MSBuild/Visual Studio: set the property `VcpkgAdditionalInstallOptions` to `--debug`
+
+## Troubleshooting binary caching configuration issues
+
+There are many choices for a binary caching provider including AWS, NuGet, Azure. While each provider will have its own error messages, we expect the type of error and the actual fix will be the same across all providers. We will walk through a specific example of using NuGet as a binary cache provider.
+
+### Binary package ABI
+
+The full Application Binary Interface (ABI) of the binary package is in the debug output. For example, for zlib:
+
+```
+[DEBUG] Trying to hash <path>\buildtrees\zlib\x86-windows.vcpkg_abi_info.txt
+[DEBUG] <path>\buildtrees\zlib\x86-windows.vcpkg_abi_info.txt has hash bb1c96759ac96102b4b18215db138daedd3eb16c2cd3302ae7bffab2b643eb87
+```
+
+The ABI hash `bb1c96759ac96102b4b18215db138daedd3eb16c2cd3302ae7bffab2b643eb87` for package zlib is constructed by hashing all the possible relevant information (sources, compiler information) to distinguish binary packages. The binary cache provider uses this as a key to refer to the binary package artifact. If this hash were to change between runs for the same package, it would mean some part of the package had changed.
+
+### Creating the binary package
```

Comment:
What does failure look like? The user is going to paste error messages into their search engine.